setup.py

```diff
@@ -291,17 +335,43 @@ def load(filename):
 
 
 class build_ext(build_ext_parent):
+    def get_outputs(self):
+        return [self.build_lib]
+
     def run(self):
         self.run_command('build_proto')
-        for ext in self.extensions:
-            ext.pre_run()
         return setuptools.command.build_ext.build_ext.run(self)
 
+    def build_extensions(self):
+        i = 0
+        while i < len(self.extensions):
+            ext = self.extensions[i]
+            fullname = self.get_ext_fullname(ext.name)
+            filename = self.get_ext_filename(fullname)
+
+            lib_path = CMAKE_BUILD_DIR
+            if os.name == 'nt':
+                debug_lib_dir = os.path.join(lib_path, "Debug")
+                release_lib_dir = os.path.join(lib_path, "Release")
+                if os.path.exists(debug_lib_dir):
+                    lib_path = debug_lib_dir
+                elif os.path.exists(release_lib_dir):
+                    lib_path = release_lib_dir
+            src = os.path.join(lib_path, filename)
+
+            if not os.path.exists(src):
```

Comment:
We don't need this logic here, because the extensions list is fixed in onnx (aka. we are always building ['onnx_cpp2py_export']).